docs/developer-guide/image-build.md

```diff
@@ -1,21 +1,345 @@
+    * **bpfman-agent**: Agent that listens to KubeAPI Server and makes calls to bpfman to load or unload
+      eBPF programs based on user intent.
+    * **bpfman-operator**: Operator for deploying bpfman.
+    * **tc-dispatcher**: eBPF container image containing the TC Dispatcher, which is used by bpfman
+      to manage and allow multiple TC based programs to be loaded on a given TC hook point.
+    * **xdp-dispatcher**: eBPF container image containing the XDP Dispatcher, which is used by bpfman
+      to manage and allow multiple TC based programs to be loaded on a given XDP hook point.
+    * **csi-node-driver-registrar**: CSI Driver used by bpfman.
+    * **bpfman-operator-bundle**: Image containing all the CRDs (Custom-Resource-Definitions) used
+      by bpfman-agent to define Kubernetes objects used to manage eBPF programs.
+* [quay.io/bpfman-bytecode](https://quay.io/organization/bpfman-bytecode): This repository contains
+  eBPF container images for all of the generated bytecode from 
+  [examples/](https://github.com/bpfman/bpfman/tree/main/examples/) and
+  [integration-test/](https://github.com/bpfman/bpfman/tree/main/tests/integration-test/bpf).
+* [quay.io/bpfman-userspace](https://quay.io/organization/bpfman-userspace):  This repository contains
+  userspace container images for all of the example programs in
+  [examples/](https://github.com/bpfman/bpfman/tree/main/examples/).
 
-### bpfman
+## Multiple Architecture Support
+
+All `bpfman` related container images that are automatically built and pushed to `quay.io/` contain a manifest file
+and images built for the following architectures:
+
+* x86_64
+* arm64
+* ppc64le
+* s390x
+
+## Locally Build bpfman-operator and bpfman-agent Container Images
+
+When testing or developing in bpfman-operator, it may be necessary to run with updated changes
+to the bpfman-operator or bpfman-agent container images.
+The local Makefile will build and load both images based on the current changes: 
 
 ```sh
-docker build -f /Containerfile.bpfman . -t bpfman:local
+cd $HOME/src/bpfman-operator/
+
+make build-images
+make run-on-kind
 ```
 
-## Running locally in container
+## Locally Build bpfman Container Image
+
+When testing or developing in bpfman-operator, it may be necessary to run with updated changes
+to bpfman.
+By default, bpfman-agent uses `quay.io/bpfman/bpfman:latest`.
+To build the bpfman binaries in a container image, run:
 
-### bpfman
+```sh
+cd $HOME/src/bpfman/
+
+docker build -f ./Containerfile.bpfman.local . -t quay.io/$QUAY_USER/bpfman:test
+```
+
+Use any registry, image name and tag, above is just an example.
+Next, build and deploy the bpfman-operator and bpfman-agent with the locally built bpfman container
+image.
 
 ```sh
-sudo docker run --init --privileged --net=host -v /etc/bpfman/certs/:/etc/bpfman/certs/ -v /sys/fs/bpf:/sys/fs/bpf quay.io/bpfman/bpfman:latest
+cd $HOME/src/bpfman-operator/
+
+BPFMAN_IMG=quay.io/$QUAY_USER/bpfman:test make build-images
```

Comment:
Actually, `make build-images` is not necessary because it is run on `make run-on-kind`. I tend to do it first in my development so that if there is an error, it gets caught earlier. I'll just remove this line.

I also scrubbed everywhere BPFMAN_IMG was used. See testing.md chnages.